lib/engine/game/g_steam_over_holland/step/buy_train.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/buy_train'
+
+module Engine
+  module Game
+    module GSteamOverHolland
+      module Step
+        class BuyTrain < Engine::Step::BuyTrain
+          def actions(entity)
+            return [] if entity != current_entity
+
+            return ['sell_shares'] if entity == current_entity&.owner && can_ebuy_sell_shares?(current_entity)
+            return %w[sell_shares buy_train pass] if president_may_contribute?(entity)
+            return %w[buy_train pass] if can_buy_train?(entity)
+
+            []
+          end
+
+          def pass_description
+            if can_close_corp?(current_entity)
+              "Close #{current_entity.name}"
+            else
+              @acted ? 'Done (Trains)' : 'Skip (Trains)'
+            end
+          end
+
+          def process_pass(action)
+            entity = action.entity
+
+            raise GameError, 'Corporation can afford train, a train must be purchased.' if can_afford_needed_train?(entity)
```

Comment:
This is where you need to check if the corporation has a route, and only throw an error if there is one.
To avoid the graph being calculated during game load, structure the test along the lines of:
```raise GameError '…' if !@game.loading && has_route && can_afford```
That means it will get checked when a player is making a move, but when the game is being loaded it is assumed that this has already been checked and is OK.

Reply:
Yep, had to flip it around into if... raise GameError, but it's done.

Reply:
Whoops, that should definitely be !@game.loading 

Fixing now...